app/services/email_output_service.rb

```diff
@@ -0,0 +1,70 @@
+class EmailOutputService
+  def initialize(email_service:)
+    @email_service = email_service
+  end
+
+  def execute(submission_id:, action:, attachments:, pdf_attachment:)
+    email_attachments = []
+
+    if action.fetch(:include_attachments) == true
+      email_attachments = email_attachments.concat attachments
+    end
+    if action.fetch(:include_pdf) == true
+      email_attachments << pdf_attachment
+    end
+
+    if email_attachments.empty?
+      send_single_email(
+        action: action,
+        attachments: [],
```

Comment:
If email_attachments is empty, do we need to do this check?
Can we not just pass the empty attachments and have a single function work on either a list or empty list?

Reply:
that's how i stated this but this seemed simpler in the end